src/js/covidMap.js

```diff
@@ -437,6 +537,97 @@ fluid.defaults("fluid.covidMap.map", {
                     record: "fluid.covidMap.filterCheckboxInMap"
                 }
             }
+        },
+        desktopViewHandler: {
+            source: "{map}.model.isDesktopView",
+            type: "fluid.covidMap.desktopViewHandler"
+        },
+        mobileViewHandler: {
+            source: "{map}.model.isMobileView",
+            type: "fluid.covidMap.mobileViewHandler"
+        }
+    }
+});
+
+fluid.covidMap.detectWindowSize = function (that) {
+    const mediaQuery = window.matchMedia(that.options.mediaQueryBreakpoint);
+    that.applier.change("isDesktopView", mediaQuery.matches ? true : false);
+    mediaQuery.addListener(function (evt) {
+        that.applier.change("isDesktopView", evt.matches ? true : false);
+    });
+};
+
+// The desktop view handler
+fluid.defaults("fluid.covidMap.desktopViewHandler", {
+    gradeNames: "fluid.modelComponent",
+    listeners: {
+        "onCreate.setVisiblePanels": {
+            listener: "{map}.applier.change",
+            args: ["visiblePanels", ["citiesList", "filterPanel"]]
+        },
+        "onCreate.setExpandedPanels": {
+            listener: "{map}.applier.change",
+            args: ["expandedPanels", {
+                citiesList: true,
+                resultsPage: true,
+                filterPanel: true,
+                hospitalPanel: true
+            }]
+        }
+    },
+    modelRelay: {
+        citiesShowing: {
+            source: "resultsShowing",
+            target: "dom.citiesList.visible",
+            func: x => !x
+        }
+    }
+});
+
+// The mobile view handler
+fluid.defaults("fluid.covidMap.mobileViewHandler", {
+    gradeNames: "fluid.modelComponent",
+    listeners: {
+        "onCreate.setVisiblePanels": {
+            listener: "{map}.applier.change",
+            args: ["visiblePanels", ["citiesList"]]
+        },
+        "onCreate.setExpandedPanels": {
+            listener: "{map}.applier.change",
+            args: ["expandedPanels", {
+                citiesList: false,
+                resultsPage: true,
+                filterPanel: true,
+                hospitalPanel: true
+            }]
+        }
+    },
+    modelRelay: {
+        // handle "back to location" buttons
+        backToCities: {
+            source: "{map}.model.query",
+            target: "{map}.model.visiblePanels",
+            func: x => x ? ["resultsPage"] : ["citiesList"]
+        },
+        onlyShowResultsPage: {
+            source: "{map}.model.resultsShowing",
+            target: "{map}.model.visiblePanels",
+            func: x => {if (x) {return ["resultsPage"];}}
+        },
+        expandResultsPage: {
+            source: "{map}.model.resultsShowing",
+            target: "{map}.model.expandedPanels.resultsPage",
+            func: x => {if (x) {return true;}}
+        },
+        onlyShowHospitalPanel: {
+            source: "{map}.model.selectedIndex",
```

Comment:
It's somewhat hard to follow what is going on here... especially since there is a different style of representation for the expansion and visibility of the panels. The comment at the head of this block doesn't seem fully accurate. 
It looks something like "When there are results, expand and show the resultsPage only"?